internal/broker/broker.go

```diff
@@ -398,86 +398,88 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	if err = b.updateSession(sessionID, session); err != nil {
 		return AuthDenied, "", err
 	}
+
+	encoded, err := json.Marshal(iaResponse)
+	if err != nil {
+		return AuthDenied, "", fmt.Errorf("could not parse data to JSON: %v", err)
+	}
+
+	data := string(encoded)
+	if data == "null" {
+		data = `""`
```

Comment:
**One thing, in authd's `brokers/broker.go`  we have:

```go
if data == "" {
		data = "{}"
	}
```

Isn't this breaking such assumption? And we may return a string instead of an object?**

Reply:
Good point, actually. I think it's then better to keep it as an empty string then and only run `json.Valid` in cases where `data != ""`, wdyt? 

I was thinking about formatting all the empty strings as well, but it's not very idiomatic Go to not return empty values when you have an error.

Reply:
Imho we should just rerun an empty object and that's it... So it's valid and we don't have to use the brokers check. As ideally it should assume that the returned json is valid